lib/modules/manager/gomod/artifacts.ts

```diff
@@ -275,11 +287,11 @@ export async function updateArtifacts({
     logger.trace({ cmd, args }, 'go get command included');
     execCommands.push(`${cmd} ${args}`);
 
-    // Update import paths on major updates above v1
+    // Update import paths on major updates
     const isImportPathUpdateRequired =
       config.postUpdateOptions?.includes('gomodUpdateImportPaths') &&
-      config.updateType === 'major' &&
-      config.newMajor! > 1;
+      config.updateType === 'major';
```

Comment:
By removing this logic here and into later, you end up with `isImportPathUpdateRequired` being `true` in some cases when it was `false`. Is the result still the same, that for newMajor <=1 no commands are run?

Reply:
This case is moved into the filter function you commented on above. So as long as all the upgrades are either from gopkg.in or going from v0 to v1 no `mod upgrade ...` commands are run.